### PR TITLE
Feature/question details component

### PR DESCRIPTION
**Pull request by rachaelcarroll**
## Description

I built out the basic structure for the QuestionDetails component and commented how we need to map through a question id's matching answers to create a `div` for each one to live in our QuestionDetails container.  Each should have a timestamp (maybe? if so, we need to do a conversion to display this) and I think it would be a neat addition if there was a "like" or "upvote" button icon.  Each time this is clicked, we would need to do a new POST to that answer to only increment the rating. But that can be a future problem :) 

I also updated the render in App where we will conditionally render QuestionDetails based on the /question/:id path.  We need to look into updating the path on the Card component (when clicked it should Route to /questions/:id and that should solve the issue)

I made comments on some possible solutions throughout. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor

## Concerns
Some next steps:  
- Once our backend is deployed we need to update the URL in our fetch questions... create a GET answers and POST answers in apiCalls
- Pass in the question id of the specified/clicked question card into our fetch call for the answer 
- Work on the CSS throughout

### CODE REVIEW

File: src/Components/QuestionDetails.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+
+const QuestionDetails = ({id, questions, answers}) => {
+
+    const displayedQuestion = questions.find(question => question.id === id)
+    
+    //we need to somehow pass in JUST the answers for that question id into props ... map through them to create a separate
```
Comment (andrewvallejo) [inline src/Components/QuestionDetails.js:9]: Would be easier once we have data for the answers. I guess a way to practice this until the server is done would be to make stubs for the answers.

Comment (shayan-golafshani) [reply src/Components/QuestionDetails.js:9]: Depending on how we're passing down the id, we may need to parseInt!

File: src/Components/QuestionDetails.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    //div for each one. then we can insert them in our return in line 25 (probably not the correct layout, but just giving you the general idea)
+    const allAnswers = answers.map(answer => {
+        return (
+            <article className='answer'>
+                <div className='time-rating'>
+                    <p>{answer.time_stamp}</p>
+                    <button className='rating'>{answer.rating} likes</button>
+                </div>
+                <p>{answer.answer}</p>
+            </article>
```
Comment (shayan-golafshani) [inline src/Components/QuestionDetails.js:19]: this looks solid, I would just consider re-aligning the button tags for clarity.

**Review by shayan-golafshani — APPROVED** [review]
Looks great Rachel, we can try to add in hooks later.

**Pull request merged**